### PR TITLE
Defined reCAPTCHA config.

### PR DESCRIPTION
**Pull request by Xiaoshouzi-gh**
- Added reCAPTCHA protection states.
 - Added reCAPTCHA action rule.
 - Added reCAPTCHA key config.

No tests needed for this change.

Hey there! So you want to contribute to a Firebase SDK? 
Before you file this pull request, please read these guidelines:

### Discussion

  * Read the contribution guidelines (CONTRIBUTING.md).
  * If this has been discussed in an issue, make sure to link to the issue here. 
    If not, go file an issue about this **before creating a pull request** to discuss.

### Testing

  * Make sure all existing tests in the repository pass after your change.
  * If you fixed a bug or added a feature, add a new test to cover your code.

### API Changes

  * At this time we cannot accept changes that affect the public API.  If you'd like to help 
    us make Firebase APIs better, please propose your change in an issue so that we 
    can discuss it together.

### CODE REVIEW

File: src/auth/recaptcha-config.ts [hunk 1]
```diff
@@ -0,0 +1,80 @@
+/*!
+ * Copyright 2020 Google Inc.
```
Comment (renkelvin) [inline src/auth/recaptcha-config.ts:2]: 2022

**Review by renkelvin — COMMENTED** [review]
It seems better to move the configs to `auth-config.ts` as other configs there.

File: src/auth/auth-config.ts [hunk 2]
```diff
@@ -1451,3 +1451,68 @@ export class OIDCConfig implements OIDCAuthProviderConfig {
...
+  */
+ endScore: number;
+  /**
+  * The action for reCAPTCHA-protected requests.
+  */
+ action?: RecaptchaAction;
+}
+
+/**
+ * The key's platform type: only web supported now..
```
Comment (lahirumaramba) [inline src/auth/auth-config.ts:1484]: nit: extra `.`

**Comment by lahirumaramba** [comment]
Thanks, @Xiaoshouzi-gh !
LGTM. Before you merge, please get a TW review on the public docs.

Is the plan to update tenant and project config interfaces in a separate PR?

File: src/auth/auth-config.ts (hunk 2 above)
Comment (Xiaoshouzi-gh) [reply src/auth/auth-config.ts]: - Fixed extra `.`.  
- As of the TW review, we were planning to have a TW to do a full review when all pieces in the reCaptcha is ready to be merge into the master branch.
- Yes, the plan is the update the tenant and project config in a separate PR.

File: src/auth/recaptcha-config.ts (hunk 1 above)
Comment (Xiaoshouzi-gh) [reply src/auth/recaptcha-config.ts]: Moved  reCaptcha config to auth-config.  This issue does not applicable anymore.

File: src/auth/auth-config.ts (hunk 2 above)
Comment (lahirumaramba) [reply src/auth/auth-config.ts]: Sounds good. Thank you!

**Review by lahirumaramba — APPROVED** [review]

**Pull request merged**